Modules/timemodule.c

```diff
@@ -656,6 +690,45 @@ time_strftime(PyObject *self, PyObject *args)
     }
 #endif
 
+#ifdef MS_WINDOWS
+    /*For Windows firstly replace %Z with time zone name from Windows API
+    to not use format string containing %Z with wcsftime().*/
+    wchar_t *result, *tmp;
+    const wchar_t *ins;
+    wchar_t zone[128];
+    size_t len_zone, len_copy, count;
+
+    get_windows_zone(zone);
+    len_zone = wcslen(zone);
+
+    // Count the number of %Z occurences
+    ins = fmt;
+    for (count = 0; tmp = wcsstr(ins, L"%Z"); ++count) {
+        ins = tmp + 2;
+    }
+
+    //Replace %Z with time zone name
+    if (count) {
+        size_t l = wcslen(fmt) + (len_zone - 2) * count + 1;
+        tmp = result = (time_char *)PyMem_Malloc(l * sizeof(time_char));
+        while (count--) {
+            ins = wcsstr(fmt, L"%Z");
+            len_copy = ins - fmt;
+            if (wcsncmp(ins - 1, L"%", 1) == 0) {
```

Comment:
If the format starts with a "%Z", `ins == fmt` and `ins - 1` is outside of a buffer.

If "%Z" follows a double "%" ("%%%Z"), it is copied as is.